heaps/min_heap.py

```diff
@@ -59,20 +65,41 @@ def heap_up(self, index):
             property is reestablished.
             
             This could be **very** helpful for the add method.
-            Time complexity: ?
-            Space complexity: ?
+            Time complexity: O(log n)
+            Space complexity: O(1)
         """
-        pass
+        parent_node = (index-1) // 2
+
+        if index:
+            if self.store[index].key < self.store[parent_node].key:
```

Comment:
Could also be written

```suggestion
        if index and self.store[index].key < self.store[parent_node].key:
```